apis/elbv2/v1alpha1/custom_types.go

```diff
@@ -0,0 +1,142 @@
+	FixedResponseConfig *FixedResponseActionConfig `json:"fixedResponseConfig,omitempty"`
+	// Information about a forward action.
+	ForwardConfig *CustomForwardActionConfig `json:"forwardConfig,omitempty"`
+
+	// The order for the action. This value is required for rules with multiple
+	// actions. The action with the lowest value for order is performed first.
+	Order *int64 `json:"order,omitempty"`
+	// Information about a redirect action.
+	//
+	// A URI consists of the following components: protocol://hostname:port/path?query.
+	// You must modify at least one of the following components to avoid a redirect
+	// loop: protocol, hostname, port, or path. Any components that you do not modify
+	// retain their original values.
+	//
+	// You can reuse URI components using the following reserved keywords:
+	//
+	//    * #{protocol}
+	//
+	//    * #{host}
+	//
+	//    * #{port}
+	//
+	//    * #{path} (the leading "/" is removed)
+	//
+	//    * #{query}
+	//
+	// For example, you can change the path to "/new/#{path}", the hostname to "example.#{host}",
+	// or the query to "#{query}&value=xyz".
+	RedirectConfig *RedirectActionConfig `json:"redirectConfig,omitempty"`
+
+	// The Amazon Resource Name (ARN) of the target group. Specify only when
+	// actionType is forward and you want to route to a single target group.
+	// To route to one or more target groups, use ForwardConfig instead.
+	// +optional
+	TargetGroupARN *string `json:"targetGroupARN,omitempty"`
+
+	// Reference to TargetGroupARN used to set TargetGroupARN
+	// +optional
+	TargetGroupARNRef *xpv1.Reference `json:"targetGroupARNRef,omitempty"`
+
+	// Selector for references to TargetGroups for TargetGroupARNs
+	// +optional
+	TargetGroupARNSelector *xpv1.Selector `json:"targetGroupARNSelector,omitempty"`
+
+	// The type of action.
+	// +kubebuilder:validation:Required
+	Type *string `json:"actionType"` // renamed json tag from "type_"
+}
+
+// CustomListenerParameters includes the custom fields of Listener.
+type CustomListenerParameters struct {
+	// The actions for the default rule.
+	// +kubebuilder:validation:Required
+	DefaultActions []*CustomAction `json:"defaultActions"`
+
+	// The Amazon Resource Name (ARN) of the load balancer.
+	// +optional
+	LoadBalancerARN *string `json:"loadBalancerARN,omitempty"`
+
+	// Ref to loadbalancer ARN
+	// +optional
+	LoadBalancerARNRef *xpv1.Reference `json:"loadBalancerARNRef,omitempty"`
+
+	// Selector for references to LoadBalancer for LoadBalancerARN
+	// +optional
+	LoadBalancerARNSelector *xpv1.Selector `json:"loadBalancerARNSelector,omitempty"`
+}
+
+// CustomLoadBalancerParameters includes the custom fields of LoadBalancer.
+type CustomLoadBalancerParameters struct {
+	// The type of load balancer. The default is application.
+	Type *string `json:"loadBalancerType,omitempty"`
+
+	// Reference to Security Groups for SecurityGroups field
+	// +optional
+	SecurityGroupsRef []xpv1.Reference `json:"securityGroupsRef,omitempty"`
```

Comment:
`SecurityGroupRefs`

We usually go for plural in Refs instead of the field in case of arrays.